script.js

```diff
@@ -1,4 +1,210 @@
+/*
+Requirements (Base)
+1. There are 2 players and players take turns 
+
+2. When a player clicks Submit, the game rolls 2 dice and shows the dice rolls, for example 3 and 6.
+3. The player picks the order of the dice they want. For example, if they wanted the number 63, they would specify that the 2nd dice goes first. You can choose how the player specifies dice order.
+4. After both players have rolled and chosen dice order, the player with the higher combined number wins.
+
+
+concepts involved
+1. Game states
+2. Loops
+3. Arrays 
+4. Loops w Arrays?
+5. Can Boolean be used? 
+
+
+how to 
+- must have mechanism to move from player 1 to player 2
+- 
+*/
+
+// Create variables for the game modes
+var player1Roll = "player 1 roll dices";
+var player1Select = "player 1 select dice order";
+
+var player2Roll = "player 2 roll dices";
+var player2Select = "player 2 select dice order";
+
+var matchPlayer1Player2 = "Check who wins";
+
+var player1FirstRollOutput = 0;
+var player1SecondRollOutput = 0;
+var player2FirstRollOutput = 0;
+var player2SecondRollOutput = 0;
+
+var player1Array = [];
+var player2Array = [];
+
+var player1Number = 0;
+var player2Number = 0;
+
+// set to true if the game is over
+var isTheGameOver = false;
+
+var selectDiceOrderText =
+  "Choose the order of the dice. Type '1' for first dice first. Alternatively, type '2' for second dice first.";
+
+// Set current game mode so user will roll dice when click submit
+var currentGameMode = player1Roll;
+
+//create function for dice roll
+var diceRoll = function () {
+  return Math.floor(Math.random() * 6) + 1;
+};
+
+//create function emcompassing all game modes & game logic
+var beatThatGame = function (input) {
+  // game mode: player 1 to roll
+  if (currentGameMode == player1Roll) {
+    if (input != "") {
+      myOutputValue = "Please roll the dice by clicking 'submit'";
+      return myOutputValue;
```

Comment:
good input validation, i know we taught the use of the variable `myOutputValue` at the start but this was just to make it understand what the variable is for but actually, you can just return the string directly so this can be simplified to `return "Please roll the dice by clicking 'submit'"`